### PR TITLE
open up ParallelLoadOperation

### PR DESCRIPTION
**Pull request by cdietrich**
open up ParallelLoadOperation
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/resourceloader/ParallelResourceLoader.java [hunk 1]
```diff
@@ -185,7 +185,7 @@ public void run() {
 
 				// push resource to the queue, wait if queue is full
 				try {
-					resourceQueue.put(Tuples.create(uri, resource, exception));
+					getResourceQueue().put(Tuples.create(uri, resource, exception));
```
Comment (cdietrich) [inline org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/resourceloader/ParallelResourceLoader.java:188]: am not sure if this has performance impact if called 10ks of times

File: org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/resourceloader/ParallelResourceLoader.java [hunk 2]
```diff
@@ -198,8 +198,8 @@ public LoadResult next() {
 				throw new NoSuchElementException("The resource queue is empty or the execution was cancelled.");
 			Triple<URI, Resource, Throwable> result = null;
 			try {
-				result = resourceQueue.poll(waitTime, TimeUnit.MILLISECONDS);
-				toProcess--;
+				result = getResourceQueue().poll(waitTime, TimeUnit.MILLISECONDS);
+				setToProcess(getToProcess() - 1);
```
Comment (cdietrich) [inline org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/resourceloader/ParallelResourceLoader.java:202]: same here

File: org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/resourceloader/ParallelResourceLoader.java [hunk 3]
```diff
@@ -231,19 +231,19 @@ public LoadResult next() {
 
 		@Override
 		public boolean hasNext() {
-			return toProcess > 0;
+			return getToProcess() > 0;
```
Comment (cdietrich) [inline org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/resourceloader/ParallelResourceLoader.java:234]: same here

File: org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/resourceloader/ParallelResourceLoader.java [hunk 4]
```diff
@@ -231,19 +231,19 @@ public LoadResult next() {
...
 		@Override
 		public void load(Collection<URI> uris) {
 			synchronizeResources(uris);
 			
-			toProcess += uris.size();
+			setToProcess(getToProcess() + uris.size());
 			Collection<URI> workload = getSorter().sort(uris);
 			for(URI uri : workload) {
-				executor.execute(new ResourceLoadJob(uri));
+				getExecutor().execute(new ResourceLoadJob(uri));
```
Comment (cdietrich) [inline org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/resourceloader/ParallelResourceLoader.java:244]: and here

File: org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/resourceloader/ParallelResourceLoader.java (hunk 1 above)
Comment (szarnekow) [reply org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/resourceloader/ParallelResourceLoader.java:188]: Good question. My gut feeling is that hotspot will inline the field access.

**Pull request closed without merging**